insights/core/__init__.py

```diff
@@ -284,6 +285,36 @@ def get(self, item, default=None):
         return self.data.get(item, default)
 
 
+def get_xmlns(self, root):
+        return root.tag.strip("{").split("}")[0] if all(c in root.tag for c in ["{", "}"]) else ""
+
+
+class XMLParser(Parser):
+    """
+    A parser class that reads XML files.  Base your own parser on this.
+    """
+
+    def parse_content(self, content):
```

Comment:
As a public `parse_content` method of the base class, better implement it a bit comment to avoid the sub-classes to re-write it.

Here is an example:
```python
    def _parse_dom(self, dom):
        return {}

    def parse_content(self, content):
        # ignore empty xml file
        self.dom = self.xmlns = None
        self.data = {}
        if len(content) > 3:
            root = ET.fromstring('\n'.join(content))
            self.dom = root
            self.data = self._parse_dom(root)
            self.xmlns = self.get_xmlns(root)
```

BTW, better add the empty content check to the base class to avoid duplicate it in sub-classes.

Reply:
Sorry, _parse_dom(self, dom) should be _parser_dom(self). `dom` should not be passed in